python/lsst/scarlet/lite/models/free_form.py

```diff
@@ -19,19 +19,20 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-__all__ = ["FreeFormComponent"]
+__all__ = ["FactorizedFreeFormComponent"]
 
-from typing import cast
+from typing import Callable, cast
 
 import numpy as np
 
 from ..bbox import Box
-from ..component import FactorizedComponent
+from ..component import Component, FactorizedComponent
 from ..detect import footprints_to_image
-from ..parameters import Parameter
+from ..image import Image
+from ..parameters import Parameter, parameter
 
 
-class FreeFormComponent(FactorizedComponent):
+class FactorizedFreeFormComponent(FactorizedComponent):
```

Comment:
A complete rename of a class like this should at least be a deprecation with an alias until it's removed, if not an RFC.

Reply:
Normally I would agree but nothing under `models` has ever been used in production (and still isn't with this ticket). AFAIK no one other than me and a grad student that I work with have ever used this code, as it's all been experimental up until now. Deprecation also doesn't work, because I used the name for an actual FreeFormComponent (with no spectral constraints).

Reply:
As I wrote in #software-dev on LSSTC, you should be able to do this:
`
from deprecated.sphinx import deprecated

class FactorizedFreeFormComponent(FactorizedComponent):
    ...

@deprecated(reason="Renamed to FactorizedFreeFormComponent", version="v28.0", category=FutureWarning)
class FreeFormComponent(FactorizedFreeFormComponent):
    pass
`
It's up to you but seeing as this class has been on main for over a year, I think it's best to deprecate the name properly.

Reply:
I understand your point and normally I would just deprecate it (since it isn't that hard). But in this case I actually reuse the name for a general, non-factorized FreeFormComponent (see https://github.com/lsst/scarlet_lite/blob/tickets/DM-41840/python/lsst/scarlet/lite/models/free_form.py#L143-L242). Had this been in broader use I probably would use a temporary name for the new class, with a deprecation warning on both, but I feel pretty strongly that in this instance no one even knows about these classes other than me and it's only a minor inconvenience to someone if I'm wrong.